pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -92,61 +92,60 @@ func (rc *reconciler) reconciliationLoopFunc() func() {
 				if err != nil {
 					// Skip detaching this volume if unable to update node status
-					glog.Infof("UpdateNodeStatuses failed with: %v", err)
+					glog.Errorf("UpdateNodeStatuses failed while attempting to report volume %q as attached to node %q with: %v ",
+						err,
+						attachedVolume.VolumeName,
+						attachedVolume.NodeName)
 					continue
 				}
 
-				if !attachedVolume.MountedByNode {
-					glog.V(5).Infof("Attempting to start DetachVolume for volume %q from node %q", attachedVolume.VolumeName, attachedVolume.NodeName)
-					err := rc.attacherDetacher.DetachVolume(attachedVolume.AttachedVolume, true /* verifySafeToDetach */, rc.actualStateOfWorld)
-					if err == nil {
-						glog.Infof("Started DetachVolume for volume %q from node %q", attachedVolume.VolumeName, attachedVolume.NodeName)
-					}
-					if err != nil &&
-						!nestedpendingoperations.IsAlreadyExists(err) &&
-						!exponentialbackoff.IsExponentialBackoff(err) {
-						// Ignore nestedpendingoperations.IsAlreadyExists && exponentialbackoff.IsExponentialBackoff errors, they are expected.
-						// Log all other errors.
-						glog.Errorf(
-							"operationExecutor.DetachVolume failed to start for volume %q (spec.Name: %q) from node %q with err: %v",
-							attachedVolume.VolumeName,
-							attachedVolume.VolumeSpec.Name(),
-							attachedVolume.NodeName,
-							err)
-					}
-				} else {
-					// If volume is not safe to detach (is mounted) wait a max amount of time before detaching anyway.
-					if timeElapsed > rc.maxWaitForUnmountDuration {
-						glog.V(5).Infof("Attempting to start DetachVolume for volume %q from node %q. Volume is not safe to detach, but maxWaitForUnmountDuration expired.", attachedVolume.VolumeName, attachedVolume.NodeName)
-						err := rc.attacherDetacher.DetachVolume(attachedVolume.AttachedVolume, false /* verifySafeToDetach */, rc.actualStateOfWorld)
-						if err == nil {
-							glog.Infof("Started DetachVolume for volume %q from node %q due to maxWaitForUnmountDuration expiry.", attachedVolume.VolumeName, attachedVolume.NodeName)
-						}
-						if err != nil &&
-							!nestedpendingoperations.IsAlreadyExists(err) &&
-							!exponentialbackoff.IsExponentialBackoff(err) {
-							// Ignore nestedpendingoperations.IsAlreadyExists && exponentialbackoff.IsExponentialBackoff errors, they are expected.
-							// Log all other errors.
-							glog.Errorf(
-								"operationExecutor.DetachVolume failed to start (maxWaitForUnmountDuration expiry) for volume %q (spec.Name: %q) from node %q with err: %v",
-								attachedVolume.VolumeName,
-								attachedVolume.VolumeSpec.Name(),
-								attachedVolume.NodeName,
-								err)
-						}
-					}
+				// Trigger detach volume which requires verifing safe to detach step
+				// If timeout is true, skip verifySafeToDetach check
+				verifySafeToDetach := !timeout
+				if timeout {
+					glog.Errorf("Volume %q is not safe to detach, but maxWaitForUnmountDuration %v expired",
```

Comment:
Add node name to message.

Alternatively, and I'll leave it up to you to decide, but I would combine this with the `Started DetachVolume for volume` message below and just print one instead two separate log entries. (I'd like to be able to search for "Started DetachVolume for" strings and get all the information I need).